src/app/kubernetes/components/pipeline-status/pipeline-status.component.scss

```diff
@@ -1,382 +1,554 @@
+  }
+  to {
+    background-color: transparent;
+  }
 }
 
-.pipeline-circle {
-    background: #d1d1d1;
-    width: 18px;
-    height: 18px;
-    border-radius: 9px;
-    margin-top: -11px;
-    position: relative;
-    transform: rotate(-90deg)
+// Solid circle backgrounds
+$keyframes fadeInSuccess {
+  to {background-color: $build-pipeline-success-color;}
 }
 
-.pipeline-circle:after {
-    position: absolute;
-    color: #fff;
-    font-family: FontAwesome;
-    font-size: 12px;
-    transform: translate(-50%,-50%) rotate(90deg);
-    top: 50%;
-    left: 50%;
-    opacity: 0;
-    animation: fadeIn .1s 875ms linear forwards
+$keyframes fadeInFailed {
+  to {background-color: $build-pipeline-failed-color;}
 }
 
-.pipeline-circle .clip1:before,.pipeline-circle .clip2:before {
-    width: 18px;
-    height: 18px;
-    transform: rotate(360deg);
-    border-radius: 9px;
-    position: absolute;
-    content: ''
+$keyframes fadeInProgress {
+  to {background-color: $build-pipeline-in-progress-color;}
 }
 
-.pipeline-circle .clip1 {
-    position: absolute;
-    clip: rect(0,18px,18px,9px);
-    z-index: -9
+$keyframes fadeInAborted {
+  to {background-color: $build-pipeline-aborted-color;}
 }
 
-.pipeline-circle .clip1:before {
-    clip: rect(0,9px,18px,0);
-    animation: progress 175ms .35s linear forwards
+$keyframes fadeInNotExecuted {
+  to {background-color: $build-pipeline-pending-color;}
 }
 
-.pipeline-circle .clip2 {
-    position: absolute;
-    clip: rect(0,9px,18px,0);
-    z-index: -9
+.build-pipeline {
+  border: 1px solid $build-pipeline-border-color;
+  border-top-width: 2px;
+  &:first-child {
+    border-top-width: 1px;
+  }
 }
 
-.pipeline-circle .clip2:before {
-    clip: rect(0,18px,18px,9px);
-    animation: progress 175ms 525ms linear forwards
+.pipeline-container {
+  @mixin flex($columns: 1 1 auto);
+  overflow: hidden;
 }
-
-.pipeline-circle .inner-circle {
-    height: 10px;
-    width: 10px;
-    border-radius: 9px;
+.pipeline {
+  @mixin flex-display($display: flex);
+  @mixin flex-wrap($wrap: wrap);
+  height: 100%;
+  padding: 0 ($pipeline-padding / 2);
+  .overview & {
     background-color: #fff;
-    position: absolute;
-    top: 4px;
-    left: 4px;
-    animation: fadeOut .1s .7s linear forwards
-}
-
-.pipeline-circle .inner-circle .inner-circle-fill {
-    box-sizing: border-box;
-    height: 100%;
+  }
+  .pipeline-stage {
+    // need to use flex-basis: auto and width because of IE11 bug
+    @mixin flex($columns: 0 0 auto);
+    min-height: 96px;
+    padding: ($pipeline-padding + 5) $pipeline-padding  ($pipeline-padding * 3) $pipeline-padding;
+    position: relative;
     width: 100%;
-    border-radius: 50%;
-    opacity: 0
-}
-
-
-
-.pipeline-status-bar .clip1:before,.pipeline-status-bar .clip2:before,.pipeline-status-bar .pipeline-line:before {
-    background-color: #d1d1d1
-}
-
-.pipeline-status-bar .inner-circle-fill {
+    // add arrow after each stage
```

Comment:
this should be an icon rather than a unicode value:

Up: `<i class="fa fa-arrow-up" aria-hidden="true"></i>`
Down: `<i class="fa fa-arrow-down" aria-hidden="true"></i>`